### PR TITLE
Correct `MappingProxyType` for 3.8+

### PR DESCRIPTION
**Pull request by max-muoto**
It seems in earlier versions of Python, stubtest wasn't able to correctly introspect on `MappingProxyType`, leading to inaccurate stubs. Unlike on `Mapping`, `get` should only accept positional arguments.

### CODE REVIEW

File: stdlib/types.pyi [hunk 1]
```diff
@@ -304,6 +304,10 @@ class MappingProxyType(Mapping[_KT, _VT_co]):
     def keys(self) -> KeysView[_KT]: ...
     def values(self) -> ValuesView[_VT_co]: ...
     def items(self) -> ItemsView[_KT, _VT_co]: ...
+    @overload
+    def get(self, key: _KT, /) -> _VT_co | None: ...  # type: ignore[override]
+    @overload
+    def get(self, key: _KT, default: _T2, /) -> _VT_co | _T2: ...  # type: ignore[override]
```
Comment (hauntsaninja) [inline stdlib/types.pyi:310]:
```suggestion
    def get(self, key: _KT, default: _VT_co | _T2, /) -> _VT_co | _T2: ...  # type: ignore[override]
```
I don't remember why `Mapping.get` is typed the way it is, but maybe we should match that over here

Comment (max-muoto) [reply stdlib/types.pyi]: Yeah, I don't think it makes a difference, at least on newer versions of MyPy/Pyright, but probably good to stay consistent for now.

Comment (max-muoto) [reply stdlib/types.pyi]: Hmm, starting to see some failures now.

**Comment by hauntsaninja** [comment]
Nice, it gets rid of the primer hits. Unfortunately needs a stubtest ignore...

**Comment by max-muoto** [comment]
> Nice, it gets rid of the primer hits. Unfortunately needs a stubtest ignore...

Ah, looks like we have one on `dict.get` anyway. Let me add it in.

File: stdlib/@tests/stubtest_allowlists/common.txt [hunk 2]
```diff
@@ -111,6 +111,7 @@ tkinter.simpledialog.TclVersion
 tkinter.simpledialog.TkVersion
 tkinter.Misc.after  # we intentionally don't allow everything that "works" at runtime
 traceback.TracebackException.from_exception  # explicitly expanding arguments going into TracebackException __init__
+types.MappingProxyType.get
```
Comment (max-muoto) [inline stdlib/@tests/stubtest_allowlists/common.txt:114]: dict is under the "should fix" category, so putting it here as well. I think we should revisit both ignores at some point.

**Review by hauntsaninja — APPROVED** [review]
Thanks!

**Pull request merged**